core/src/main/java/fj/data/Either.java

```diff
@@ -724,11 +752,7 @@ public boolean exists(final F<B, Boolean> f) {
   public static <A, X> Either<List<A>, X> sequenceLeft(final List<Either<A, X>> a) {
     return a.isEmpty() ?
            Either.<List<A>, X>left(List.<A>nil()) :
-           a.head().left().bind(new F<A, Either<List<A>, X>>() {
-             public Either<List<A>, X> f(final A aa) {
-               return sequenceLeft(a.tail()).left().map(cons_(aa));
-             }
-           });
+           a.head().left().bind(aa -> sequenceLeft(a.tail()).left().map(cons_(aa)));
```

Comment:
Note the separate calls to sequenceLeft and sequenceRight for my comment on traverseLeftList/traverseRightList.